### DISCUSSION TITLE
Should I enable NVIDIA GPU or is it not worth it?

### DISCUSSION THREAD

**Discussion by slopokdave** [immich-app/immich#6234] [Q&A]
I've got a 1660 Super so completely capable, but I'm confused a bit on if I lose other features if I go the Nvidia route:

- Do I have to use the "NOML" branch, therefore I lose ML? So my search will be slower? What about facial recognition features, do I lose those too?
- Is hardware encoding stable with Immich? 

Thanks, really enjoying Immich so far. :)

**Comment by AngelaDMerkel** [comment]
Hardware acceleration is useful, especially for transcoding and it does not come at the cost of any features. I am not using hardware encoding for Immich because everything is lightning quick using CPU as is, and the rare time I upload a video, ffmpeg produces better results for streaming to clients through CPU encoding. 
An initial upload of thousands of images may take two or three days, but honestly, that does not matter when it's realistically likely to happen only once.
